app/models/on_duty_locator.rb

```diff
@@ -0,0 +1,22 @@
+class OnDutyLocator
+  def initialize(time, rota_slots)
+    @time = time
+    @rota_slots = rota_slots
+  end
+
+  def locate
+    on_duty_organisation_uid
+  end
+
+  private
+
+  attr_reader :time, :rota_slots
+
+  def slot_with_on_duty_shift
+    rota_slots.detect { |slot| slot.shift.starting_time < time }
```

Comment:
Does this heavily depend on the order in which the `rota_slots` are passed into this class?